aspnetcore/tutorials/first-mvc-app/adding-model.md

```diff
@@ -99,278 +108,275 @@ The preceding command adds the [aspnet-codegenerator scaffolding tool](xref:fund
+[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/StartupDevProd.cs?name=snippet_StartupClass&highlight=3,5,10,16-28)]
 
-<a name="cs"></a>
+# [Visual Studio for Mac](#tab/visual-studio-mac)
 
-## Add a database connection string
+Open a command window in the project directory. The project directory is the directory that contains the *Program.cs*, *Startup.cs*, and *.csproj* files.
 
-Add a connection string to the *appsettings.json* file:
+Export the scaffold tool path:
 
-# [Visual Studio](#tab/visual-studio)
+```console
+export PATH=$HOME/.dotnet/tools:$PATH
+```
 
-[!code-json[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/appsettings.json?highlight=10-12)]
+Run the following command:
 
-# [Visual Studio Code / Visual Studio for Mac](#tab/visual-studio-code+visual-studio-mac)
+```dotnetcli
+dotnet-aspnet-codegenerator controller -name MoviesController -m Movie -dc MvcMovieContext --relativeFolderPath Controllers --useDefaultLayout --referenceScriptLibraries -sqlite
 
-[!code-json[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/appsettings_SQLite.json?highlight=10-12)]
+```
 
----
+[!INCLUDE [explains scaffold generated params](~/includes/mvc-intro/model4.md)]
 
-Build the project as a check for compiler errors.
+<a name="sqlite-dev-vs-mac"></a>
 
-## Scaffold movie pages
+### Use SQLite for development, SQL Server for production
 
-Use the scaffolding tool to produce Create, Read, Update, and Delete (CRUD) pages for the movie model.
+When SQLite is selected, the template generated code is ready for development. The following code shows how to inject <xref:Microsoft.AspNetCore.Hosting.IWebHostEnvironment> into `Startup`. `IWebHostEnvironment` is injected so `ConfigureServices` can use SQLite in development and SQL Server in production.
 
-# [Visual Studio](#tab/visual-studio)
+[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/StartupDevProd.cs?name=snippet_StartupClass&highlight=3,5,10,16-28)]
 
-In **Solution Explorer**, right-click the *Controllers* folder **> Add > New Scaffolded Item**.
+---
 
-![view of above step](adding-model/_static/add_controller21.png)
+<!-- End of tabs                  -->
 
-In the **Add Scaffold** dialog, select **MVC Controller with views, using Entity Framework > Add**.
+<a name="scaffolding-created"></a>
 
-![Add Scaffold dialog](adding-model/_static/add_scaffold21.png)
+Scaffolding creates:
 
-Complete the **Add Controller** dialog:
+* A movies controller: *Controllers/MoviesController.cs*
+* Razor view files for Create, Delete, Details, Edit, and Index pages: `Views/Movies/*.cshtml`
+* A database context class: *Data/MvcMovieContext.cs*
 
-* **Model class:** *Movie (MvcMovie.Models)*
-* **Data context class:** *MvcMovieContext (MvcMovie.Data)*
+The automatic creation of these files is known as *scaffolding*.
```

Comment:
Scaffolding updates Startup, appSetting.json, and the project file.  anything else?

Reply:
@Rick-Anderson,  will put in what you suggested, but will move this section to tabs.
I had assumed the reason you originally kept this list to just the files created was that you were applying it to all three tabs, and it was too much detail for here.  VS does it all for scaffolding, but VS Code requires steps you take in addition to running dotnet aspnet-codegenerator controller.  So in the VS Code context I will list what scaffolding creates, but I need to exclude updates to appsettings.json for the connection string etc.  which were done manually, and not by scaffolding.

Reply:
Resolved with e5c4187